components/lock.rst

```diff
@@ -0,0 +1,306 @@
+| :ref:`FlockStore <lock-store-flock>`         | local  | yes      | no        |
++----------------------------------------------+--------+----------+-----------+
+| :ref:`MemcachedStore <lock-store-memcached>` | remote | no       | yes       |
++----------------------------------------------+--------+----------+-----------+
+| :ref:`RedisStore <lock-store-redis>`         | remote | no       | yes       |
++----------------------------------------------+--------+----------+-----------+
+| :ref:`SemaphoreStore <lock-store-semaphore>` | local  | yes      | no        |
++----------------------------------------------+--------+----------+-----------+
+
+.. tip::
+
+    Calling the :method:`Symfony\\Component\\Lock\\LockInterface::refresh`
+    method on a Lock created from a non expirable ``Store`` like
+    :ref:`FlockStore <lock-store-flock>` will do nothing.
+
+.. _lock-store-flock:
+
+FlockStore
+~~~~~~~~~~
+
+The FlockStore use the fileSystem on the local computer to lock and store the
+``resource``. It does not supports expiration, but the lock is automatically
+released when the PHP process is terminated.
+
+.. code-block:: php
+
+    use Symfony\Component\Lock\Store\FlockStore;
+
+    $store = new FlockStore(sys_get_temp_dir());
+
+The first argument of the constructor is the path to the directory where the
+file will be created.
+
+.. caution::
+
+    Beware, some filesystem (like some version of NFS) does not support locking.
+    We suggest to use local file, or to use a Store dedicated to remote usage
+    like Redis or Memcached.
+
+.. _Packagist: https://packagist.org/packages/symfony/lock
+
+.. _lock-store-memcached:
+
+MemcachedStore
+~~~~~~~~~~~~~~
+
+The MemcachedStore stores state of ``resource`` in a Memcached server. This
+``Store`` does not support blocking, and expect a TLL to avoid infinity locks.
+
+.. note::
+
+    Memcached does not supports TTL lower than 1 seconds.
+
+It requires to have installed Memcached and have created a connection that
+implements the ``\Memcached`` classes::
+
+.. code-block:: php
+
+    use Symfony\Component\Lock\Store\RedisStore;
+
+    $memcached = new \Memcached();
+    $memcached->addServer('localhost', 11211);
+
+    $store = new MemcachedStore($memcached);
+
+.. _lock-store-redis:
+
+RedisStore
+~~~~~~~~~~
+
+The RedisStore uses an instance of Redis to store the state of the ``resource``.
+This ``Store`` does not support blocking, and expect a TLL to avoid infinity
+locks.
+
+It requires to have installed Redis and have created a connection that
+implements the ``\Redis``, ``\RedisArray``, ``\RedisCluster`` or ``\Predis``
+classes::
+
+.. code-block:: php
+
+    use Symfony\Component\Lock\Store\RedisStore;
+
+    $redis = new \Redis();
+    $redis->connect('localhost');
+
+    $store = new RedisStore($redis);
+
+.. _lock-store-semaphore:
+
+SemaphoreStore
+~~~~~~~~~~~~~~
+
+The SemaphoreStore uses the PHP semaphore function to lock a ``resources``.
```

Comment:
You're right, I was thinking about `PHP semaphore function**s**` is it more clear? Or should I rewrite the sentence?